hypothesis-python/docs/how-to/type-strategies.rst

```diff
@@ -0,0 +1,57 @@
+Write type hints for strategies
+===============================
+
+Hypothesis provides type hints for all standard strategies:
+
+.. code-block:: python
+
+    from hypothesis import strategies as st
+
+    # SearchStrategy[int]
+    reveal_type(st.integers())
+
+    # SearchStrategy[list[int]]
+    reveal_type(st.lists(st.integers()))
+
+|SearchStrategy| is the type of a strategy. You can use it to write type hints for your own strategies:
+
+.. code-block:: python
+
+    from hypothesis import strategies as st
+    from hypothesis.strategies import SearchStrategy
+
+
+    @st.composite
+    def all_ints_or_floats(draw) -> SearchStrategy[int] | SearchStrategy[float]:
+        all_integers = draw(st.booleans())
+        return st.integers() if all_integers else st.floats()
+
+Type variance of |SearchStrategy|
+---------------------------------
+
+|SearchStrategy| is generic in the type of examples it generates. It is covariant, meaning that if ``B < A`` then ``SearchStrategy[B] < SearchStrategy[A]``. In other words:
+
+.. code-block:: python
+
+    from hypothesis import given, strategies as st
+
+
+    class A:
+        pass
+
+
+    class B(A):
+        pass
```

Comment:
I like to use classes `Animal` and `Dog` without showing the definitions; it's clear and saves some space.  Also unclear whether it's worth writing out the full imports every time 🤷‍♂️

Reply:
I wanted to show where to import `SearchStrategy` from because I don't think it's obvious that it's from `st.` and not top level, and at that point not including `from hypothesis import strategies as st` felt weird. agree that in general this late in the docs we can generally drop imports